crates/uv/src/lib.rs

```diff
@@ -42,8 +44,50 @@ pub(crate) mod printer;
 pub(crate) mod settings;
 pub(crate) mod version;
 
+/// Resolves the script target for a run command.
+///
+/// If it's a local file, does nothing. If it's a URL, downloads the content
+/// to a temporary file and updates the command. Prioritizes local files over URLs.
+/// Returns Some(NamedTempFile) if a remote script was downloaded, None otherwise.
+async fn resolve_script_target(
+    command: &mut ExternalCommand,
+) -> Result<Option<tempfile::NamedTempFile>> {
+    let Some(target) = command.get_mut(0) else {
+        return Ok(None);
+    };
+
+    // Only continue if we are absolutely certain no local file exists.
+    if let Ok(false) = Path::new(target).try_exists() {
+    } else {
+        return Ok(None);
+    }
```

Comment:
I think you can just do:

```rust
if !matches!(Path::new(target).try_exists(), Ok(false)) {
    return Ok(None);
}
```

There are actually a lot of different ways to write this, but I think I like this one the best. But the `if` with the empty body is a bit odd. :)

Reply:
ooo, I like this.